src/compiler/checker.ts

```diff
@@ -4564,10 +4588,14 @@ namespace ts {
                 propTypes.push(type);
             }
             const result = <TransientSymbol>createSymbol(SymbolFlags.Property | SymbolFlags.Transient | SymbolFlags.SyntheticProperty | commonFlags, name);
+            result.syntheticKind = SyntheticSymbolKind.UnionOrIntersection;
             result.containingType = containingType;
             result.hasNonUniformType = hasNonUniformType;
             result.isPartial = isPartial;
             result.declarations = declarations;
+            if (declarations.length) {
+                result.valueDeclaration = declarations[0];
+            }

```

Comment:
Checking index constraints in `checkIndexConstraints` relies on `valueDeclaration`. It's not needed now that we don't create synthetic symbols for spread any more, so I removed it. But in general `checkIndexConstraints` works on synthetic properties by mistake. If choosing an arbitrary valueDeclaration isn't right, the `checkIndexConstraints` needs to stop relying on it.